modules/dns_dnscmd/dns_dnscmd_main.rb

```diff
@@ -85,5 +85,31 @@ def report msg, response, error_only
       logger.error "Dnscmd failed:\n" + (response.is_a?(Array) ? response.join("\n") : "Response was not an array! #{response}")
       raise Proxy::Dns::Error.new("Unknown error while processing '#{msg}'")
     end
+
+    def match_zone(fqdn)
+      dns_zones = ::Proxy::Dns::Plugin.settings.dns_zones
```

Comment:
Let me rephrase this.  I think this should be a required property -- otherwise it's possible to skip this config setting and end up with the issue that you found and fixed.